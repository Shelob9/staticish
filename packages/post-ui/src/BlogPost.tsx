import React from 'react';

import { Post } from '@staticish/wp-api-to-static/';

export default function BlogPost(props: { post: Post }) {
  const { post } = props;
  function createContent() {
    return { __html: post.content.rendered };
  }

  return (
    <div>
      <h1>{post.title.rendered}</h1>
      <div dangerouslySetInnerHTML={createContent()} />
    </div>
  );
}
