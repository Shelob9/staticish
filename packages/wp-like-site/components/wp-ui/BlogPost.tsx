import React, { Fragment } from "react";
import Article from "./Primative/Article";
import Footer from "./Primative/Footer";
import PostTags from "./PostTags";
import PostContent, { renderPropForContent } from "./PostContent";
import FeaturedImage, { renderFeaturedImage } from "./FeaturedImage";
import PostHeader from "./PostHeader";
import { PostAuthor, Image } from "./types";
import { WpPost, WpAuthor } from "./wpTypes";
import Author from "./Author";

//Copypasta!
const addAuthorLink = (author: WpAuthor) => {
	return {
		...author,
		link: {
			href: `/authors/${author.slug}`,
			text: `Read More`,
			title: `All Posts By ${author.name}`
		}
	};
};

function PostTitle(props: { title: string }) {
	return (
		<PostHeader
			level={1}
			tw="font-bold font-sans break-normal text-gray-900 pt-6 pb-2 text-3xl md:text-4xl"
			title={props.title}
		/>
	);
}

/**
 * Props for the "renderTitle" render prop.
 */
export type renderTitleProps = {
	title: string;
	post: WpPost;
};
type renderPropForTitle = (props: renderTitleProps) => JSX.Element;

export function PostTop(props: {
	post: WpPost;
	renderTitle?: (props: renderTitleProps) => JSX.Element;
	editMode?: boolean;
	onClickSettings?: () => void;
	renderBelowTitle?: () => JSX.Element;
	renderFeatured?: renderFeaturedImage;
}) {
	const { post, onClickSettings, renderFeatured } = props;
	const { title, published } = post;
	const featured: Image = post.featured
		? post.featured
		: {
				src: "http://placekitten.com/1280/640",
				alt: "A cat"
		  };
	return (
		<Fragment>
			<FeaturedImage featured={featured} renderFeatured={renderFeatured} />
			<div
				className="w-full px-4 md:px-6 text-xl text-gray-800 leading-normal"
				style={{ fontFamily: "Georgia,serif" }}
			>
				<div className="font-sans">
					<div className="text-base md:text-sm text-teal-500 font-bold inline-block">
						<span>
							{props.renderTitle ? (
								props.renderTitle({
									title: title.rendered,
									post
								})
							) : (
								<PostTitle title={title.rendered} />
							)}
							{props.renderBelowTitle ? (
								React.createElement(props.renderBelowTitle, {
									post
								})
							) : (
								<p className="text-sm md:text-base font-normal text-gray-600">
									Published {published}
								</p>
							)}
						</span>
					</div>
					{onClickSettings && (
						<div className={"inline-block"}>
							<button onClick={onClickSettings}>Settings</button>
						</div>
					)}
				</div>
			</div>
		</Fragment>
	);
}
export default function(props: {
	post: WpPost;
	renderTitle?: renderPropForTitle;
	renderContent?: renderPropForContent;
	onClickSettings?: () => void;
	renderBelowTitle?: () => JSX.Element;
	renderFeatured?: renderFeaturedImage;
}) {
	const {
		post,
		renderTitle,
		renderContent,
		onClickSettings,
		renderBelowTitle,
		renderFeatured
	} = props;
	const { tags, author, content } = post;
	return (
		<Article tw="container w-full md:max-w-3xl mx-auto pt-20">
			<hr className="border-b-2 border-gray-400 mb-8 mx-4" />
			<PostTop
				post={props.post}
				renderTitle={renderTitle}
				onClickSettings={onClickSettings}
				renderBelowTitle={renderBelowTitle}
				renderFeatured={renderFeatured}
			/>
			<PostContent content={content.rendered} renderContent={renderContent} />
			<Footer>
				{undefined !== tags && tags.length && (
					<div className="text-base md:text-sm text-gray-500 px-4 py-6">
						<PostTags tags={tags} />
					</div>
				)}
				<Author author={addAuthorLink(author)} />
			</Footer>
		</Article>
	);
}
