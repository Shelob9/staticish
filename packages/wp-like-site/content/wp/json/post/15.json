{"id":15,"date":"2019-12-26T18:09:36","date_gmt":"2019-12-26T18:09:36","guid":{"rendered":"http://localhost:3100/?p=15"},"modified":"2019-12-26T18:09:36","modified_gmt":"2019-12-26T18:09:36","slug":"other-author","status":"publish","type":"post","link":"http://localhost:3100/other-author/","title":{"rendered":"Other Author"},"content":{"rendered":"\n<h2> Are et dolore magna aliqua ?</h2>\n\n\n\n<p>Are et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.  </p>\n","protected":false},"excerpt":{"rendered":"","protected":false},"author":2,"featured_media":0,"comment_status":"open","ping_status":"open","sticky":false,"template":"","format":"standard","meta":[],"categories":[5],"tags":[6,7],"_links":{"self":[{"href":"http://localhost:3100/wp-json/wp/v2/posts/15"}],"collection":[{"href":"http://localhost:3100/wp-json/wp/v2/posts"}],"about":[{"href":"http://localhost:3100/wp-json/wp/v2/types/post"}],"author":[{"embeddable":true,"href":"http://localhost:3100/wp-json/wp/v2/users/2"}],"replies":[{"embeddable":true,"href":"http://localhost:3100/wp-json/wp/v2/comments?post=15"}],"version-history":[{"count":1,"href":"http://localhost:3100/wp-json/wp/v2/posts/15/revisions"}],"predecessor-version":[{"id":16,"href":"http://localhost:3100/wp-json/wp/v2/posts/15/revisions/16"}],"wp:attachment":[{"href":"http://localhost:3100/wp-json/wp/v2/media?parent=15"}],"wp:term":[{"taxonomy":"category","embeddable":true,"href":"http://localhost:3100/wp-json/wp/v2/categories?post=15"},{"taxonomy":"post_tag","embeddable":true,"href":"http://localhost:3100/wp-json/wp/v2/tags?post=15"}],"curies":[{"name":"wp","href":"https://api.w.org/{rel}","templated":true}]}}