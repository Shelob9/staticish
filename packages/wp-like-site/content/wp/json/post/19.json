{"id":19,"date":"2019-12-26T18:11:37","date_gmt":"2019-12-26T18:11:37","guid":{"rendered":"http://localhost:3100/?p=19"},"modified":"2019-12-26T18:11:47","modified_gmt":"2019-12-26T18:11:47","slug":"cats","status":"publish","type":"post","link":"http://localhost:3100/cats/","title":{"rendered":"Cats"},"content":{"rendered":"\n<figure class=\"wp-block-gallery aligncenter columns-2 is-cropped\"><ul class=\"blocks-gallery-grid\"><li class=\"blocks-gallery-item\"><figure><img src=\"http://localhost:3100/wp-content/uploads/2019/12/cat-statue.jpg\" alt=\"A statue of a cat\" data-id=\"10\" data-full-url=\"http://localhost:3100/wp-content/uploads/2019/12/cat-statue.jpg\" data-link=\"http://localhost:3100/post-about-cat/cat-statue/\" class=\"wp-image-10\" srcset=\"http://localhost:3100/wp-content/uploads/2019/12/cat-statue.jpg 800w, http://localhost:3100/wp-content/uploads/2019/12/cat-statue-300x300.jpg 300w, http://localhost:3100/wp-content/uploads/2019/12/cat-statue-150x150.jpg 150w, http://localhost:3100/wp-content/uploads/2019/12/cat-statue-768x768.jpg 768w\" sizes=\"(max-width: 800px) 100vw, 800px\" /><figcaption class=\"blocks-gallery-item__caption\">Statue Kitten</figcaption></figure></li><li class=\"blocks-gallery-item\"><figure><img src=\"http://localhost:3100/wp-content/uploads/2019/12/tiny-cat.jpg\" alt=\"A very tiny cat\" data-id=\"7\" data-full-url=\"http://localhost:3100/wp-content/uploads/2019/12/tiny-cat.jpg\" data-link=\"http://localhost:3100/post-with-cat-as-featured-image/tiny-cat/\" class=\"wp-image-7\"/><figcaption class=\"blocks-gallery-item__caption\">Kitten Kitten</figcaption></figure></li></ul></figure>\n\n\n\n<figure class=\"wp-block-image size-large\"><img src=\"http://localhost:3100/wp-content/uploads/2019/12/cat-statue.jpg\" alt=\"\" class=\"wp-image-10\" srcset=\"http://localhost:3100/wp-content/uploads/2019/12/cat-statue.jpg 800w, http://localhost:3100/wp-content/uploads/2019/12/cat-statue-300x300.jpg 300w, http://localhost:3100/wp-content/uploads/2019/12/cat-statue-150x150.jpg 150w, http://localhost:3100/wp-content/uploads/2019/12/cat-statue-768x768.jpg 768w\" sizes=\"(max-width: 800px) 100vw, 800px\" /><figcaption>Cat statue</figcaption></figure>\n\n\n\n<h2> Are et dolore magna aliqua ?</h2>\n\n\n\n<p>Are et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.  </p>\n","protected":false},"excerpt":{"rendered":"","protected":false},"author":1,"featured_media":0,"comment_status":"open","ping_status":"open","sticky":false,"template":"","format":"standard","meta":[],"categories":[3],"tags":[4],"_links":{"self":[{"href":"http://localhost:3100/wp-json/wp/v2/posts/19"}],"collection":[{"href":"http://localhost:3100/wp-json/wp/v2/posts"}],"about":[{"href":"http://localhost:3100/wp-json/wp/v2/types/post"}],"author":[{"embeddable":true,"href":"http://localhost:3100/wp-json/wp/v2/users/1"}],"replies":[{"embeddable":true,"href":"http://localhost:3100/wp-json/wp/v2/comments?post=19"}],"version-history":[{"count":1,"href":"http://localhost:3100/wp-json/wp/v2/posts/19/revisions"}],"predecessor-version":[{"id":20,"href":"http://localhost:3100/wp-json/wp/v2/posts/19/revisions/20"}],"wp:attachment":[{"href":"http://localhost:3100/wp-json/wp/v2/media?parent=19"}],"wp:term":[{"taxonomy":"category","embeddable":true,"href":"http://localhost:3100/wp-json/wp/v2/categories?post=19"},{"taxonomy":"post_tag","embeddable":true,"href":"http://localhost:3100/wp-json/wp/v2/tags?post=19"}],"curies":[{"name":"wp","href":"https://api.w.org/{rel}","templated":true}]}}