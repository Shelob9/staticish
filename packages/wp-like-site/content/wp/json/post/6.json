{"id":6,"date":"2019-12-26T18:07:27","date_gmt":"2019-12-26T18:07:27","guid":{"rendered":"http://localhost:3100/?p=6"},"modified":"2019-12-26T18:07:27","modified_gmt":"2019-12-26T18:07:27","slug":"post-with-cat-as-featured-image","status":"publish","type":"post","link":"http://localhost:3100/post-with-cat-as-featured-image/","title":{"rendered":"Post With Cat As Featured Image"},"content":{"rendered":"\n<blockquote class=\"wp-block-quote\"><p>Lab Rats Are Only Powerless Because They Don’t Understand That They’re In An Experiment</p><cite>&#8211; The OA</cite></blockquote>\n\n\n\n<p> Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labor.</p>\n\n\n\n<h2>Heading</h2>\n\n\n\n<p> re et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.  </p>\n\n\n\n<p> re et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.  </p>\n","protected":false},"excerpt":{"rendered":"<p>Lab Rats Are Only Powerless Because They Don’t Understand That They’re In An Experiment &#8211; The OA Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labor. Heading re et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. [&hellip;]</p>\n","protected":false},"author":1,"featured_media":7,"comment_status":"open","ping_status":"open","sticky":false,"template":"","format":"standard","meta":[],"categories":[3],"tags":[2],"_links":{"self":[{"href":"http://localhost:3100/wp-json/wp/v2/posts/6"}],"collection":[{"href":"http://localhost:3100/wp-json/wp/v2/posts"}],"about":[{"href":"http://localhost:3100/wp-json/wp/v2/types/post"}],"author":[{"embeddable":true,"href":"http://localhost:3100/wp-json/wp/v2/users/1"}],"replies":[{"embeddable":true,"href":"http://localhost:3100/wp-json/wp/v2/comments?post=6"}],"version-history":[{"count":1,"href":"http://localhost:3100/wp-json/wp/v2/posts/6/revisions"}],"predecessor-version":[{"id":8,"href":"http://localhost:3100/wp-json/wp/v2/posts/6/revisions/8"}],"wp:featuredmedia":[{"embeddable":true,"href":"http://localhost:3100/wp-json/wp/v2/media/7"}],"wp:attachment":[{"href":"http://localhost:3100/wp-json/wp/v2/media?parent=6"}],"wp:term":[{"taxonomy":"category","embeddable":true,"href":"http://localhost:3100/wp-json/wp/v2/categories?post=6"},{"taxonomy":"post_tag","embeddable":true,"href":"http://localhost:3100/wp-json/wp/v2/tags?post=6"}],"curies":[{"name":"wp","href":"https://api.w.org/{rel}","templated":true}]}}