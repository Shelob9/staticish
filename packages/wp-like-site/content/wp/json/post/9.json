{"id":9,"date":"2019-12-26T18:08:36","date_gmt":"2019-12-26T18:08:36","guid":{"rendered":"http://localhost:3100/?p=9"},"modified":"2019-12-26T18:08:36","modified_gmt":"2019-12-26T18:08:36","slug":"post-about-cat","status":"publish","type":"post","link":"http://localhost:3100/post-about-cat/","title":{"rendered":"Post ABout Cat"},"content":{"rendered":"\n<figure class=\"wp-block-image size-large\"><img src=\"http://localhost:3100/wp-content/uploads/2019/12/cat-statue.jpg\" alt=\"\" class=\"wp-image-10\" srcset=\"http://localhost:3100/wp-content/uploads/2019/12/cat-statue.jpg 800w, http://localhost:3100/wp-content/uploads/2019/12/cat-statue-300x300.jpg 300w, http://localhost:3100/wp-content/uploads/2019/12/cat-statue-150x150.jpg 150w, http://localhost:3100/wp-content/uploads/2019/12/cat-statue-768x768.jpg 768w\" sizes=\"(max-width: 800px) 100vw, 800px\" /><figcaption>A statue of a cat</figcaption></figure>\n\n\n\n<h2> Are et dolore magna aliqua ?</h2>\n\n\n\n<p>Are et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.  </p>\n","protected":false},"excerpt":{"rendered":"","protected":false},"author":1,"featured_media":0,"comment_status":"open","ping_status":"open","sticky":false,"template":"","format":"standard","meta":[],"categories":[3],"tags":[4],"_links":{"self":[{"href":"http://localhost:3100/wp-json/wp/v2/posts/9"}],"collection":[{"href":"http://localhost:3100/wp-json/wp/v2/posts"}],"about":[{"href":"http://localhost:3100/wp-json/wp/v2/types/post"}],"author":[{"embeddable":true,"href":"http://localhost:3100/wp-json/wp/v2/users/1"}],"replies":[{"embeddable":true,"href":"http://localhost:3100/wp-json/wp/v2/comments?post=9"}],"version-history":[{"count":1,"href":"http://localhost:3100/wp-json/wp/v2/posts/9/revisions"}],"predecessor-version":[{"id":12,"href":"http://localhost:3100/wp-json/wp/v2/posts/9/revisions/12"}],"wp:attachment":[{"href":"http://localhost:3100/wp-json/wp/v2/media?parent=9"}],"wp:term":[{"taxonomy":"category","embeddable":true,"href":"http://localhost:3100/wp-json/wp/v2/categories?post=9"},{"taxonomy":"post_tag","embeddable":true,"href":"http://localhost:3100/wp-json/wp/v2/tags?post=9"}],"curies":[{"name":"wp","href":"https://api.w.org/{rel}","templated":true}]}}