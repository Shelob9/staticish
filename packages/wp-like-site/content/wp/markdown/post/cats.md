---
title: Cats
slug: cats
date: '2019-12-26T18:11:37'
modified: '2019-12-26T18:11:47'
excerpt: ''
---

-   ![A statue of a cat](http://localhost:3100/wp-content/uploads/2019/12/cat-statue.jpg)

    Statue Kitten

-   ![A very tiny cat](http://localhost:3100/wp-content/uploads/2019/12/tiny-cat.jpg)

    Kitten Kitten

![](http://localhost:3100/wp-content/uploads/2019/12/cat-statue.jpg)

Cat statue

## Are et dolore magna aliqua ?

Are et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
