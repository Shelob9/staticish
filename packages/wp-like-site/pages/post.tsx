import React, { Fragment } from "react";
import Head from "next/head";
import { Post } from "@staticish/wp-api-to-static";
import { BlogPostPreview } from "@staticish/post-ui";
import "../css/index.css";

export default function() {
	const post = {
		id: 168826,
		date: "2019-12-17T11:16:33",
		date_gmt: "2019-12-17T16:16:33",
		guid: { rendered: "https://calderaforms.com/?p=168826" },
		modified: "2019-12-17T11:51:49",
		modified_gmt: "2019-12-17T16:51:49",
		slug: "2019-holiday-update",
		status: "publish",
		type: "post",
		link: "https://calderaforms.com/2019/12/2019-holiday-update/",
		title: { rendered: "A Quick Holiday Update!" },
		content: {
			rendered: "\n<h2>Bug fixes as a present are kind of like getting",
			protected: false
		},
		excerpt: {
			rendered:
				'<p>Bug fixes as a present are kind of like getting socks for Christmas&#8230; not the most thrilling thing, but when you need socks&#8230; The holidays are often a different kind of time for teams. We want to keep moving all the things forward, but it&#8217;s also healthy to slow down and appreciate family, too. The &hellip; <a href="https://calderaforms.com/2019/12/2019-holiday-update/" class="more-link">Continue reading <span class="screen-reader-text">A Quick Holiday Update!</span></a></p>\n',
			protected: false
		},
		author: 62578,
		featured_media: 168887,
		comment_status: "open",
		ping_status: "open",
		sticky: false,
		template: "",
		format: "standard",
		categories: [174],
		tags: [],
		meta: {}
	};

	return (
		<Fragment>
			<Head>
				<title>Next.js + Jest + TypeScript</title>
			</Head>
			<BlogPostPreview post={post} />
		</Fragment>
	);
}
