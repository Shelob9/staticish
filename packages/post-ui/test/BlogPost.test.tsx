import React from 'react';
import BlogPost from '../src/BlogPost';
import { render } from '@testing-library/react';
import post from './test-data/post';
test('It once', () => {
  const { container } = render(<BlogPost post={post} />);
  expect(container).toMatchSnapshot();
});
