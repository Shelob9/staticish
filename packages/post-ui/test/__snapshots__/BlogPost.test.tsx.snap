// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It once 1`] = `
<div>
  <div>
    <h1>
      A Quick Holiday Update!
    </h1>
    <div>
      

      <h2>
        Bug fixes as a present are kind of like getting socks for Christmasâ€¦ not the most thrilling thing, but when you need socksâ€¦
      </h2>
      




      <p>
        The holidays are often a different kind of time for teams. We want to keep moving all the things forward, but itâ€™s also healthy to slow down and appreciate family, too. The end of the year around Saturday Drive usually isnâ€™t a time for heavy lifting, but it is a time we all devote to personal projects inside the company.
      </p>
      




      <p>
        What does this mean to you? Weâ€™ve made a point of spending a large portion of our time this holiday cooldown addressing some issues that we know have been pains in Caldera Forms for a good long while now. Hopefully below youâ€™ll find that some thorns have been removed for you!
      </p>
      




      <h2>
        Hereâ€™s what the 1.8.9 Caldera Forms update has in store for your WordPress form building experience!
      </h2>
      




      <p>
        The below fixes were selected to happen now because they were 1) bitesize enough that we could address them for you in a short period before the holidays, and 2) causing major pain for many of you. In January weâ€™ll be starting our next major build cycle, and rest assured there will be a lot more of this coming soon! 
      </p>
      




      <p>
        Hereâ€™s whatâ€™s new!
      </p>
      




      <ul>
        <li>
          Pro settings are no longer visible in the dashboard for users that are not currently connected to the Pro app. If youâ€™re a temporarily disconnected Pro user, hereâ€™s 
          <a
            href="https://calderaforms.com/doc/caldera-forms-show-pro-ui/"
          >
            how to make them visible again
          </a>
          , or 
          <a
            href="https://calderaforms.com/support/"
          >
            reach out and talk to us
          </a>
           about its modern, upgraded replacement.
        </li>
        <li>
          Fixed a PHP notice ( PHP Notice: Undefined variable: symbols in /classes/core.php on line 1416 )
        </li>
        <li>
          Completed strings that were missing from translations.
        </li>
        <li>
          Corrected the list of preset options for US states. These should now work as expected out-of-the-box.
        </li>
        <li>
          Added a {referer_url} magic tag
        </li>
        <li>
          The color picker field was fixed. You can put the crayons away now ðŸ™‚
        </li>
      </ul>
      




      <h2>
        Want a sneak preview into what January and February may bring?
      </h2>
      




      <p>
        Weâ€™re always hesitant to talk in too much detail about specific things to come. Weâ€™re always going to do our best to deliver on anything we spend time and resources on. Any developer understands that sometimes things turn out to be more complex than anticipated, take longer to complete or more resources than expected, etc etc. 
      </p>
      




      <p>
        In the spirit of better communication with all of you though, here are some highlights of what we have planned for Caldera Forms over the next couple months ðŸ™‚
      </p>
      




      <ul>
        <li>
          Expect a much better support experience soon. Weâ€™re currently hiring for an additional support person, and reworking the technical way we manage Caldera Forms support, and anticipate being able to open channels for free users soon.
        </li>
        <li>
          MailChimp will get some much-needed love. On one hand, it just plain needs fixing. On the other hand, the way it worked in the first place wasnâ€™t ideal. Weâ€™ve hatched a plot to get it into a much better place that sets a uniform stage for upgrading all other email marketing & CRM add-ons. Stretch goal: Salesforce 1.0 ðŸ™‚
        </li>
        <li>
          Many more bug fixes. Weâ€™re dedicating developers for a full-time deep dive into many more of the issues that have been a pain in your neck for way too long.
        </li>
      </ul>
      




      <h2>
        Want to chime in with bug fixes or feature requests?
      </h2>
      




      <p>
        Stay tuned. Weâ€™re working up a survey that will be going out to as many Caldera Forms users as we can reach after the turn of the year. Keep your eye out for that here and in the 
        <a
          href="https://www.facebook.com/groups/651862761663883/"
        >
          Caldera Forming Facebook group
        </a>
        . 
      </p>
      




      <p>
        Or sound off here. What bug features or fixes would you like to see in the near future?
      </p>
      

    </div>
  </div>
</div>
`;
